src/wallet/walletdb.cpp

```diff
@@ -876,31 +876,20 @@ bool CWalletDB::Recover(CDBEnv& dbenv, const std::string& filename, bool fOnlyKe
         LogPrintf("Cannot create database file %s\n", filename);
         return false;
     }
-    CWallet dummyWallet;
-    CWalletScanState wss;
 
     DbTxn* ptxn = dbenv.TxnBegin();
     BOOST_FOREACH(CDBEnv::KeyValPair& row, salvagedData)
     {
         if (fOnlyKeys)
         {
             CDataStream ssKey(row.first, SER_DISK, CLIENT_VERSION);
-            CDataStream ssValue(row.second, SER_DISK, CLIENT_VERSION);
-            string strType, strErr;
-            bool fReadOK;
-            {
-                // Required in LoadKeyMetadata():
-                LOCK(dummyWallet.cs_wallet);
-                fReadOK = ReadKeyValue(&dummyWallet, ssKey, ssValue,
```

Comment:
Not sure about this. Removing the actual deserialization routing may result in not actually checking the correctness of the data. For example, currently each `"key"` record (pub- & priv-key), is checked in `ReadKeyValue()` with `vchPubKey.IsValid()`, also, corrupt keys or a corrupt hdchain dataset will be detected now (log output) , but not after this PR.